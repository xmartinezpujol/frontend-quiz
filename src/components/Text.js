import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Text = (props) => {
  const styles = {
    h1: {
      fontSize: 20,
      color: '#36424A',
      fontWeight: 900,
      '& strong': { color: props.accentColor },
      '@media(min-width: 768px)': {
        fontSize: 24,
      },
      '@media(min-width: 992px)': {
        fontSize: 28,
      },
      '@media(min-width: 1200px)': {
        fontSize: 36,
      },
    },
    'h1.w': {
      fontSize: 20,
      color: '#FFFFFF',
      fontWeight: 900,
      '@media(min-width: 768px)': {
        fontSize: 24,
      },
      '@media(min-width: 992px)': {
        fontSize: 28,
      },
      '@media(min-width: 1200px)': {
        fontSize: 36,
      },
    },
    h2: {
      fontSize: 16,
      color: '#36424A',
      fontWeight: 900,
      '& strong': { color: props.accentColor },
      '@media(min-width: 768px)': {
        fontSize: 18,
      },
      '@media(min-width: 992px)': {
        fontSize: 22,
      },
      '@media(min-width: 1200px)': {
        fontSize: 25,
      },
    },
    'h2.1': {
      fontSize: 16,
      color: '#36424A',
      fontWeight: 700,
      '@media(min-width: 768px)': {
        fontSize: 18,
      },
      '@media(min-width: 992px)': {
        fontSize: 22,
      },
      '@media(min-width: 1200px)': {
        fontSize: 25,
      },
    },
    'h2.2': {
      fontSize: 16,
      color: '#6E7A83',
      fontWeight: 700,
      '@media(min-width: 768px)': {
        fontSize: 18,
      },
      '@media(min-width: 992px)': {
        fontSize: 22,
      },
      '@media(min-width: 1200px)': {
        fontSize: 25,
      },
    },
    'h2.w': {
      fontSize: 16,
      color: '#FFFFFF',
      fontWeight: 900,
      '@media(min-width: 768px)': {
        fontSize: 18,
      },
      '@media(min-width: 992px)': {
        fontSize: 22,
      },
      '@media(min-width: 1200px)': {
        fontSize: 25,
      },
    },
    h3: {
      fontSize: 14,
      color: '#36424A',
      fontWeight: 900,
      '& strong': { color: props.accentColor },
      '@media(min-width: 768px)': {
        fontSize: 14,
      },
      '@media(min-width: 992px)': {
        fontSize: 16,
      },
      '@media(min-width: 1200px)': {
        fontSize: 18,
      },
    },
    'h3.1': {
      fontSize: 14,
      color: '#6E7A83',
      fontWeight: 700,
      '@media(min-width: 768px)': {
        fontSize: 14,
      },
      '@media(min-width: 992px)': {
        fontSize: 16,
      },
      '@media(min-width: 1200px)': {
        fontSize: 18,
      },
    },
    'h3.w': {
      fontSize: 14,
      color: '#FFFFFF',
      fontWeight: 700,
      '@media(min-width: 768px)': {
        fontSize: 14,
      },
      '@media(min-width: 992px)': {
        fontSize: 16,
      },
      '@media(min-width: 1200px)': {
        fontSize: 18,
      },
    },
    h4: {
      fontSize: 10,
      color: '#6E7A83',
      fontWeight: 400,
      '& strong': { color: props.accentColor },
      '@media(min-width: 768px)': {
        fontSize: 12,
      },
      '@media(min-width: 992px)': {
        fontSize: 14,
      },
      '@media(min-width: 1200px)': {
        fontSize: 16,
      },
    },
    'h4.w': {
      fontSize: 10,
      color: '#FFFFFF',
      fontWeight: 400,
      '@media(min-width: 768px)': {
        fontSize: 12,
      },
      '@media(min-width: 992px)': {
        fontSize: 14,
      },
      '@media(min-width: 1200px)': {
        fontSize: 16,
      },
    },
    h5: {
      fontSize: 11,
      fontWeight: 900,
      textTransform: 'uppercase',
      '& strong': { color: props.accentColor },
      '@media(min-width: 768px)': {
        fontSize: 16,
      },
      '@media(min-width: 992px)': {

      },
      '@media(min-width: 1200px)': {

      },
    },
    'h5.w': {
      fontSize: 16,
      fontWeight: 900,
      textTransform: 'uppercase',
      color: '#FFFFFF',
    },
    p1: {
      fontSize: 16,
      color: '#36424A',
      fontWeight: 300,
      whiteSpace: 'pre-line',
      '& strong': { color: props.accentColor },
    },
    'p1.w': {
      fontSize: 16,
      color: '#FFFFFF',
      fontWeight: 300,
      whiteSpace: 'pre-line',
    },
    p2: {
      fontSize: 13,
      color: '#B6BABD',
      fontWeight: 400,
    },
    p3: {
      fontSize: 12,
      color: '#B6BABD',
      letterSpacing: '1px',
      textTransform: 'uppercase',
      fontWeight: 400,
    },
    p4: {
      fontSize: 12,
      color: '#B6BABD',
      fontWeight: 400,
    },
    p5: {
      fontSize: 12,
      color: '#6E7A83',
      fontWeight: 400,
    },
    span: {
      fontSize: 16,
      color: '#36424A',
      fontWeight: 300,
      whiteSpace: 'pre-line',
      margin: '5px 0',
    },
  };
  let propOrDefault = '';

  if (props.type.substring(0, 1) === 'h') {
    propOrDefault = typeof (styles[props.type]) !== 'undefined'
      ? props.type.substring(0, 2)
      : styles.p1;
  }
  if (props.type.substring(0, 1) === 'p') {
    propOrDefault = typeof styles[props.type] !== 'undefined'
      ? props.type.substring(0, 1)
      : styles.p1;
  }
  if (props.type.substring(0, 1) === 's') {
    propOrDefault = typeof styles[props.type] !== 'undefined'
      ? props.type.substring(0, 4)
      : styles.p1;
  }

  const TextGenerator = styled(propOrDefault)(styles[props.type]);
  return (
    <TextGenerator style={props.style}>
      {props.children}
    </TextGenerator>
  );
};

Text.propTypes = {
  accentColor: PropTypes.string,
  type: PropTypes.string.isRequired,
};

export default Text;
