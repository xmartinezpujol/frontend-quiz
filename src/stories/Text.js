import React from 'react';

import { storiesOf } from '@storybook/react';

import Text from '../components/Text';
import View from '../components/View';

storiesOf('Texts', module)
  .add('local', () => (
    <View padding={50} type="rice" direction="column">
      <Text type="h1">H1 - Im a title, trying to be descriptive!</Text>
      <Text type="h2">H2 - Im a title, trying to be descriptive!</Text>
      <Text type="h3">H3 - Im a title, trying to be descriptive!</Text>
      <Text type="h4">H4 - Im a title, trying to be descriptive!</Text>
      <Text type="h5">H4 - Im a title, trying to be descriptive!</Text>
      <Text type="p1">P1 - Im a title, trying to be descriptive!</Text>
      <Text type="p2">P2 - Im a title, trying to be descriptive!</Text>
      <Text type="p3">P3 - Im a title, trying to be descriptive!</Text>
      <Text type="p4">P4 - Im a title, trying to be descriptive!</Text>
      <Text type="p5">P5 - Im a title, trying to be descriptive!</Text>
      <Text type="span">span - Im a title, trying to be descriptive!</Text>
    </View>
  ));
