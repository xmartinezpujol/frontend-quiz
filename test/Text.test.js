import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Text from '../src/components/Text';

configure({ adapter: new Adapter() });

describe('Text', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Text type="h1">
        {"I'm a title, trying to be descriptive!"}
      </Text>,
    );
  });
  it('h1 renders correctly', () => {
    wrapper.setProps({ type: 'h1' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('h2 renders correctly', () => {
    wrapper.setProps({ type: 'h2' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('h3 renders correctly', () => {
    wrapper.setProps({ type: 'h3' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('h4 renders correctly', () => {
    wrapper.setProps({ type: 'h4' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('h5 renders correctly', () => {
    wrapper.setProps({ type: 'h5' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('p1 renders correctly', () => {
    wrapper.setProps({ type: 'p1' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('p2 renders correctly', () => {
    wrapper.setProps({ type: 'p2' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('p3 renders correctly', () => {
    wrapper.setProps({ type: 'p3' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('p4 renders correctly', () => {
    wrapper.setProps({ type: 'p4' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('p5 renders correctly', () => {
    wrapper.setProps({ type: 'p5' });
    expect(wrapper.render()).toMatchSnapshot();
  });
  it('span renders correctly', () => {
    wrapper.setProps({ type: 'span' });
    expect(wrapper.render()).toMatchSnapshot();
  });
});
