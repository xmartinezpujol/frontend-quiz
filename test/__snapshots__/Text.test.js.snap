// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text h1 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-14cjhng",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h1",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text h2 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-lysq4l",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h2",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text h3 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1ujrf7i",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h3",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text h4 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1m0rikh",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h4",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text h5 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-ard4n3",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "h5",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text p1 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-3kq6q1",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text p2 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1i2z69e",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text p3 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-1dad5y5",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text p4 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-zwpz78",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text p5 renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-16gau2z",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Text span renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "css-qg0hta",
    },
    "children": Array [
      Object {
        "data": "I'm a title, trying to be descriptive!",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
